test_runner/src/main/kotlin/ftl/run/platform/common/AfterRunTests.kt

```diff
@@ -23,7 +23,7 @@ internal suspend fun afterRunTests(
     stopwatch: StopWatch,
     config: IArgs
 ) = MatrixMap(
-    map = testMatrices.toSavedMatrixMap(),
```

Comment:
`toSavedMatrixMap` returns a mutable map. I think we could pass here immutable instead

```kotlin
private fun List<TestMatrix>.toSavedMatrixMap() =
    associate { matrix -> matrix.testMatrixId to createSavedMatrix(matrix) }.toMutableMap()
```

Reply:
toSavedMatrix now returns immutable map